malomohq/actions
# chore: unify helm upgrade command to dry env vars

# Ticket

N/A

# Motivation

Reduce repetition in CI's helm command.

Benefits:
- Reduce places that need to change when adding an env var
- Reduce potential for errors (due to forgetting to add a value to helm command)
- Simplify downstream usage of deploy action

# Changes

- use [`envsubst`](https://man7.org/linux/man-pages/man1/envsubst.1.html) to replace env vars in the values file provided to the deploy action

To test the concept locally you can run:

```bash
export SENTRY_DSN=example
envsubst < ./.k8s/values/staging.yaml
```

You should see `sentryDsn: example` in the output of `envsubst`.

# Testing Instructions/Screenshots

We could do a `--dry-run` first and test on staging to be sure.

# Dependencies

Related PR
https://github.com/malomohq/core/pull/1960

# PR Checklist

- [x] Code builds clean without any errors or warnings
- [x] I have annotated all new private and public functions with type specs
- [x] The code contains integration/unit tests when applicable
- [x] I have performed a self-review of the `Files changed` tab in this pull request
